src/layer/poi.js

```diff
@@ -0,0 +1,74 @@
+import * as label from "../constants/label.js";
+import * as Color from "../constants/color.js";
+
+export const poi = {
+  id: "poi",
+  type: "symbol",
+  paint: {
+    "text-halo-color": "#ffffff",
+    "icon-halo-color": "#ffffff",
+    "text-halo-width": 1.5,
+    "icon-halo-width": 0.4,
+    "text-halo-blur": 1,
+    "icon-halo-blur": 0.2,
+    "text-color": [
+      "match",
+      ["get", "class"],
+      ["bar", "beer"],
+      Color.poi.consumer,
+      ["hospital", "parking"],
+      Color.poi.infrastructure,
+      Color.poi.infrastructure,
+    ],
+  },
+  filter: [
+    ">=",
+    ["zoom"],
+    [
+      "match",
+      ["get", "subclass"],
+      "hospital",
+      15,
+      ["bar", "beer"],
+      16,
+      ["clinic", "doctors", "parking"],
+      17,
+      99,
+    ],
+  ],
+  layout: {
+    "text-font": ["Metropolis Light"],
+    "icon-optional": false,
+    "text-size": {
+      base: 1.0,
+      stops: [
+        [15, 10],
+        [17, 12],
+      ],
+    },
+    "icon-image": [
+      "match",
+      ["get", "subclass"],
+      ["bar", "beer"],
+      "poi_martini_glass",
+      ["clinic", "doctors"],
+      "poi_health_cross",
+      "hospital",
+      "poi_hospital",
+      "parking",
+      "poi_p",
+      "poi_square_dot", //icon for generic POI, not currently used
+    ],
```

Comment:
scripts/taginfo_template.json should contain an entry for each of the tags that correspond to the POI types added in this PR.

Ideally, taginfo.json could just run this code and automatically figure out how the `subclass`es map to icon names, but the `subclass`es differ from OSM tags to some extent. It doesn’t look like [this YAML file](https://github.com/openmaptiles/openmaptiles/blob/82b00c13897e46b9d4f39ef0a0ac39b09ebc46ad/layers/poi/poi.yaml#L19-L98) offers a way to map a `subclass` back to the original OSM tag, just the value but not the key.

Reply:
There may be some overlap between the legend’s needs and the taginfo project file’s needs.